yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -462,10 +464,23 @@ private[spark] class ApplicationMaster(
                   nextAllocationInterval = initialAllocationInterval
                   heartbeatInterval
                 }
-              logDebug(s"Number of pending allocations is $numPendingAllocate. " +
-                       s"Sleeping for $sleepInterval.")
+              sleepStart = System.currentTimeMillis()
               allocatorLock.wait(sleepInterval)
             }
+            val sleepDuration = System.currentTimeMillis() - sleepStart
+            if (sleepDuration < sleepInterval - 5) {
+              // log when sleep is interrupted
+              logInfo(s"Number of pending allocations is $numPendingAllocate. " +
```

Comment:
most users don't care about how long the internal yarn allocator loop slept for. If these logs are valuable then I would just set the `ApplicationMaster` log level to DEBUG.